src/sentry/static/sentry/app/components/projects/missingProjectMembership.tsx

```diff
@@ -47,101 +55,190 @@ class MissingProjectMembership extends React.Component<Props, State> {
       this.props.api,
       {
         orgId: this.props.organization.slug,
-        teamId: team.slug,
+        teamId: team,
       },
       {
         success: () => {
           this.setState({
             loading: false,
             error: false,
           });
+          addSuccessMessage(t('Request to join team sent.'));
         },
         error: () => {
           this.setState({
             loading: false,
             error: true,
           });
-          addErrorMessage(t('There was an error while trying to leave the team.'));
+          addErrorMessage(t('There was an error while trying to request access.'));
         },
       }
     );
   }
 
-  renderJoinTeam(team: Team, features: Set<string>) {
+  renderJoinTeam(team, features: Set<string>) {
+    const teamStatus = TeamStore.getBySlug(team);
+
     if (!team) {
       return null;
     }
     if (this.state.loading) {
-      return <a className="btn btn-default btn-loading btn-disabled">...</a>;
-    } else if (team.isPending) {
-      return <a className="btn btn-default btn-disabled">{t('Request Pending')}</a>;
+      if (features.has('open-membership')) {
+        return <StyledButton busy>Join Team</StyledButton>;
+      }
+      return <StyledButton busy>Request Access</StyledButton>;
+    } else if (teamStatus?.isPending) {
+      return <StyledButton disabled>Request Pending</StyledButton>;
     } else if (features.has('open-membership')) {
       return (
-        <a className="btn btn-default" onClick={this.joinTeam.bind(this, team)}>
-          {t('Join Team')}
-        </a>
+        <StyledButton priority="primary" onClick={this.joinTeam.bind(this, team)}>
+          Join Team
+        </StyledButton>
       );
     }
     return (
-      <a className="btn btn-default" onClick={this.joinTeam.bind(this, team)}>
-        {t('Request Access')}
-      </a>
+      <StyledButton priority="primary" onClick={this.joinTeam.bind(this, team)}>
```

Comment:
We may want a `type=button` or else clicking this button will also submit the form. (not sure what is intended here).

Reply:
Ah yeah, I removed the `Form` since it's not intended to submit a form.